edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/modules/CommonModule.cs

```diff
@@ -408,6 +417,35 @@ protected override void Load(ContainerBuilder builder)
                 .As<IClientCredentialsFactory>()
                 .SingleInstance();
 
+            // IClientCredentials "EdgeHubCredentials"
+            builder.Register(
+                    c =>
+                    {
+                        var identityFactory = c.Resolve<IClientCredentialsFactory>();
+                        IClientCredentials edgeHubCredentials = this.edgeHubConnectionString.Map(cs => identityFactory.GetWithConnectionString(cs)).GetOrElse(
+                            () => identityFactory.GetWithIotEdged(this.edgeDeviceId, this.edgeHubModuleId));
+                        return edgeHubCredentials;
+                    })
+                .Named<IClientCredentials>("EdgeHubCredentials")
+                .SingleInstance();
+
+            // ServiceIdentity "EdgeHubIdentity"
+            builder.Register(
+                    c =>
+                    {
+                        return new ServiceIdentity(
+                            this.edgeDeviceId,
+                            this.edgeHubModuleId,
+                            deviceScope: null,
+                            parentScopes: new List<string>(),
+                            this.edgeHubGenerationId.Expect(() => new InvalidOperationException("Missing generation ID")),
```

Comment:
This will throw for the connection string case. I want to know what fails if the gen id is something dummy..

Reply:
Fixed that!